src/Services/IncomeService.cs

```diff
@@ -0,0 +1,109 @@
+using Domain.DTOs;
+using Domain.Entities;
+
+using ExpenseTrackerGroup3.Repositories.Interfaces;
+using ExpenseTrackerGroup3.Services.Interfaces;
+
+namespace ExpenseTrackerGroup3.Services;
+
+public class IncomeService : IIncomeService
+{
+    private readonly IIncomeRepository _incomeRepository;
+    private readonly IUserRepository _userRepository;
+
+    public IncomeService(IIncomeRepository incomeRepository, IUserRepository userRepository)
+    {
+        _incomeRepository = incomeRepository;
+        _userRepository = userRepository;
+    }
+
+    public async Task<Income> AddIncomeAsync(Guid userId, CreateIncome income)
+    {
+        var user = await _userRepository.GetByIdAsync(userId);
+
+        if (user == null)
+        {
+            throw new ArgumentException("User not found");
+        }
+
+        var newIncome = new Income
+        {
+            Id = Guid.NewGuid(),
+            UserId = userId,
+            Amount = income.Amount,
+            Source = income.Source,
+            CreatedAt = DateTime.Now
+        };
+
+        var success = await _incomeRepository.CreateAsync(newIncome);
+
+        if (!success)
+        {
+            throw new Exception("Failed to create income");
+        }
+
+        return newIncome;
+    }
+
+    public async Task<IEnumerable<Income>> GetIncomesByUserIdAsync(Guid userId)
+    {
+
+        User? user = await _userRepository.GetByIdAsync(userId);
+
+        if (user == null) 
+        {
+            throw new ArgumentException("User not found");
+        }
+
+        IEnumerable<Income> income = await _incomeRepository.GetAllAsync();
+
+        return income.Where(i => i.UserId == userId);
+    }
+
+    public async Task<IEnumerable<Income>> GetMonthlyIncomeByUserId(Guid userId, DateTime month)
```

Comment:
This endpoint is returning incorrect data, it should return all income related to the date, but it returns all of them.

![image](https://github.com/user-attachments/assets/a2ece032-5aff-44af-aa37-2c5e202b1227)

Reply:
Yep, but, I didn't mention nothing about day, I put on Month field, 2023/09/02 `(September, 2023)`:
![image](https://github.com/user-attachments/assets/e3f0f1e6-55a0-4776-bac6-b162c3119657)

But the response is wrong, should be only first one:
![image](https://github.com/user-attachments/assets/311efde8-fbf4-4f31-bdec-0063a92c38b3)

But also are: 
![image](https://github.com/user-attachments/assets/b31c5cf6-f3cc-455e-a2e4-1c4ee4766ac9)

Reply:
Could you please give me more information about the error, because the requirement says that only the user's “income” for a specific month is retrieved?